api/v1alpha1/clientprofile_types.go

```diff
@@ -23,51 +23,60 @@ import (
 
 // CephFsConfigSpec defines the desired CephFs configuration
 type CephFsConfigSpec struct {
+	//+kubebuilder:validation:Optional
 	SubVolumeGroup string `json:"subVolumeGroup,omitempty"`
 }
 
 // RbdConfigSpec defines the desired RBD configuration
 type RbdConfigSpec struct {
+	//+kubebuilder:validation:Optional
 	RadosNamespace string `json:"radosNamespace,omitempty"`
 }
 
 // NfsConfigSpec cdefines the desired NFS configuration
 type NfsConfigSpec struct {
 }
 
-// ConfigSpec defines the desired state of Config
-type ConfigSpec struct {
+// ClientProfileSpec defines the desired state of Config
```

Comment:
// ClientProfileSpec defines the desired state of CephFS, RBD, and NFS configuration.

Reply:
done with some changes